docs/neural_modeling/plot_05_place_cells.py

```diff
@@ -0,0 +1,410 @@
+plt.xlabel("Speed (cm/s)")
+plt.ylabel("Firing rate (Hz)")
+plt.title("Neuron {}".format(neuron))
+plt.tight_layout()
+
+# %%
+# This neurons show a strong modulation of firing rate as a function of speed but we can also notice that the animal, on average, accelerates when travering the field. Is the speed tuning we observe a true modulation or spurious correlation caused by traversing the place field at different speed and for different theta phase? We can use `nemos` to model the activity and give the position, the phase and the speed as input variable.
+#
+# We will use speed, phase and position to model the activity of the neuron.
+# All the feature have already been brought to the same dimension thanks to `pynapple`.
+
+position = data["position"]
+theta = data["theta"]
+count = spikes[neuron].count(bin_size, data.time_support)
+
+print(position.shape)
+print(theta.shape)
+print(speed.shape)
+print(count.shape)
+
+# %%
+# ## Basis evaluation
+#
+# For each feature, we will use a different set of basis :
+#
+#   -   position : `nmo.basis.MSplineBasis`
+#   -   theta phase : `nmo.basis.CyclicBSplineBasis`
+#   -   speed : `nmo.basis.MSplineBasis`
+
+position_basis = nmo.basis.MSplineBasis(n_basis_funcs=10)
+phase_basis = nmo.basis.CyclicBSplineBasis(n_basis_funcs=12)
+speed_basis = nmo.basis.MSplineBasis(n_basis_funcs=15)
+
+# %%
+# In addition, we will consider position and phase to be a joint variable. In `nemos`, we can combine basis by multiplying them and adding them. In this case the final basis object for our model can be made in one line :
+
+basis = position_basis * phase_basis + speed_basis
+
+# %%
+# The object basis only tell us how each basis covers the feature space. For each timestep, we need to _evaluate_ what are the features value. We can use the `evaluate` function of `nemos`:
+
+X = basis.evaluate(position, theta, speed)
+
+# %%
+# `X` is our design matrix. For each timestamps, it contains the information about the current position, speed and theta phase of the experiment. Notice how passing a pynapple object to `evaluate` also returns a `pynapple` object.
+
+print(X)
+
+# %%
+# ## Model learning
+#
+# We can now use the Poisson GLM from nemos to learn the model.
+
+glm = nmo.glm.GLM(
+    regularizer=nmo.regularizer.UnRegularized("LBFGS", solver_kwargs=dict(tol=10**-12))
+)
+
+glm.fit(X[:, np.newaxis, :], count[:, np.newaxis])
+
+# %%
+# ## Prediction
+#
+# Let's check first if our model can accurately predict the different tuning curves we displayed above. We can use the `predict` function of nemos and then compute new tuning curves
+
+predicted_rate = glm.predict(X[:, np.newaxis, :]) / bin_size
+
+glm_pf = nap.compute_1d_tuning_curves_continuous(predicted_rate, position, 50)
+glm_pos_theta, xybins = nap.compute_2d_tuning_curves_continuous(
+    predicted_rate, data, 30
+)
+glm_speed = nap.compute_1d_tuning_curves_continuous(predicted_rate, speed, 30)
+
+# %%
+# Let's display both tuning curves together.
+
+plt.figure()
```

Comment:
maybe we should move this in the utils functions since we have other plotting code there too

Reply:
Done